TestClass.cpp

```diff
@@ -0,0 +1,234 @@
+#include <iostream>
+#include <cmath>
+#include <cassert>  
+
+using namespace std; //use std::cout
+
+#define FILE (__FILE__)
+#define LINE (__LINE__)
+
+// #define LOG_LENGTH (300)
+const int LOG_LENGTH = 300;
+
+void print_log(const char *filename,
+               int         line,
+               const char *message)
+{
+    printf("F:%s L:%d : %s\n", filename, line, message);
+}
+
+class TestClass
+{
+    private:
+        int var1;
+        int var2;
+        // double var3;
+        // double var4;
+        int max_average_value;
+        int max_square_root_value;
+
+    public:
+    //get and set functions are public
+
+        TestClass(int a, int b)
+        {
+            var1 = a;
+            var2 = b;
+            max_average_value = 10;
+            max_square_root_value = 7;
+        }
+
+#if 0
```

Comment:
I assume this if guard was used for testing using double as parameters. After the play, you can remove this extra/dead code.  Generally, if the values you use in your program are not going to be beyond the bounds of a float, no need to assign double (double uses excess accuracy/memory)

Next up, instead of if guards we can use separate files ;) and maybe combine them with a makefile in a target (consider in the future, using cmake). But that will be on a separate exercise

Reply:
The preprocessor macro will be removed in the final version. It was used to still have comments/code but not have it compiled.